source/dub/packagemanager.d

```diff
@@ -481,6 +520,10 @@ class PackageManager {
 	*/
 	int delegate(int delegate(ref Package)) getPackageIterator()
 	{
+		// See `m_initialized` documentation
+		if (!this.m_initialized)
+			this.refresh();
```

Comment:
whenever I see this pattern I think "shouldn't `refresh` handle the check"?

Reply:
`refresh` can be called multiple times and depends on the environment, so it can't decide "on its own" to skip a run.
Maybe having a simple `this.checkInitialized()` function would be a bit more self-contained, though.